qemu/tests/qemu_guest_agent.py

```diff
@@ -897,6 +897,40 @@ def gagent_check_thaw_unfrozen(self, test, params, env):
             test.fail("The return value of thawing an unfrozen fs is %s,"
                       "it should be zero" % ret)
 
+    @error_context.context_aware
+    def gagent_check_path_fsfreeze_hook(self, test, params, env):
+        """
+        Check if the default path of fsfreeze-hook is correct
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+        session = self._get_session(params, self.vm)
+        self.gagent_stop(session, self.vm)
+        error_context.context("Start gagent with -F option", logging.info)
+        params["gagent_start_cmd"] = params["gagent_start_cmd_hook"]
+        self.gagent_start(session, self.vm)
```

Comment:
I don't think you have to stop it and start it with `gagent_start_cmd_hook` again, why not directly define a different `gagent_start_cmd` in  `check_path_fsfreeze_hook` directly:
```
        - check_path_fsfreeze_hook:
            no Windows
            gagent_check_type = path_fsfreeze_hook
            gagent_help_cmd = "qemu-ga --help"
            gagent_path_cmd = "rpm -ql qemu-guest-agent"
            virtio_serial:
                gagent_start_cmd = "qemu-ga -m virtio-serial -p /dev/virtio-ports/$gagent_name -F -d"
            isa_serial:
                gagent_start_cmd = "pgrep qemu-ga || qemu-ga -d -m isa-serial -p /dev/ttyS1 -F"
```
This way you don't need L#910-L#913, and this match the case better.

Reply:
The reason of stopping it first then start it is that the qemu-guest-agent service is running in default. So I need to stop it.

Reply:
@YanhuiMa I think you need it been started by default, but just with a different `gagent_start_cmd`.

Reply:
@vivianQizhu I mean the qemu-guest-agent service is running in default when the guest is booted, not started by code.